xds/src/main/java/io/grpc/xds/internal/security/certprovider/FileWatcherCertificateProvider.java

```diff
@@ -75,14 +79,19 @@ final class FileWatcherCertificateProvider extends CertificateProvider implement
     this.timeProvider = checkNotNull(timeProvider, "timeProvider");
     this.certFile = Paths.get(checkNotNull(certFile, "certFile"));
     this.keyFile = Paths.get(checkNotNull(keyFile, "keyFile"));
-    checkArgument((trustFile != null || spiffeTrustMapFile != null),
-        "either trustFile or spiffeTrustMapFile must be present");
-    if (spiffeTrustMapFile != null) {
-      this.spiffeTrustMapFile = Paths.get(spiffeTrustMapFile);
-      this.trustFile = null;
+    if (enableSpiffe) {
```

Comment:
It looks like you don't need the flag here. With ProviderProvider checking the flag, it will guarantee `spiffeTrustMapFile == null` when `enableSpifee == false`.